user_guide.adoc

```diff
@@ -1167,25 +1243,181 @@ i.e. two least significant bits of the first byte would be lost and
-    type: u1
+  - id: packed
+    type: u2le
 instances:
-  version:
-    value: (packed_1 & 0b11110000) >> 4
-  len_header:
-    value:  packed_1 & 0b00001111
+  a:
+    value: (packed & 0b11111000_00000000) >> (3 + 8)
+  b:
+    value: (packed & 0b00000111_11111100) >> 2
+  c:
+    value:  packed & 0b00000000_00000011
+----
+
+The expressions for extracting the values look exactly the same as
+for the <<bit-ints-be, big-endian order>>, but the actual bit layout
+will be different, because here the `packed` integer is read
+in little-endian (LE) byte order.
+
+Given that `x` is a 2-byte array needed to parse an unsigned 2-byte
+integer, the numeric value of a BE integer is
+`0x{x[0]}{x[1]}` __(`0x...` is the hexadecimal integer literal
+and `{x[0]}` is the hex value of byte `x[0]` as seen in hex dumps,
+e.g. `02` or `7f`)__, whereas the value of a LE integer
+would be `0x{x[1]}{x[0]}`.
+
+This follows that if we read a BE integer from a new byte array
+`[x[1], x[0]]` (i.e. `x` reversed), we'll get the same result
+as when reading a LE integer from the original `x` array.
+
+Because we've already explained how bit-integers work in
+<<bit-ints-be, big-endian order>>, let's repeat this method for the
+above bitfield on the reversed `x` byte array (`x[1] x[0]`) and then
+swap the bytes back to the original order of `x` (`x[0] x[1]`):
+
+....
+               x[1]                              x[0]
+  7   6   5   4   3   2   1   0     7   6   5   4   3   2   1   0
+  a4  a3  a2  a1  a0  b8  b7  b6    b5  b4  b3  b2  b1  b0  c1  c0
+  ──────────────┬───────────────    ───────────────┬──────────────
+                └──────────────╷   ╷───────────────┘
+                                ╲ ╱
+                                 ╳
+                                ╱ ╲
+                ┌──────────────╵   ╵──────────────┐
+               x[0]                              x[1]
+  7   6   5   4   3   2   1   0     7   6   5   4   3   2   1   0
+  b5  b4  b3  b2  b1  b0  c1  c0    a4  a3  a2  a1  a0  b8  b7  b6
+  ──────────┬───────────  ──┬───    ─────────┬────────  ────┬─────
+            b               c                a              b
+│ <──────────────────────────── │  │ <──────────────────────────── │
+╷       parsing direction                                          ↑
+└┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈>┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┘
+....
+
+As you can guess from the bit layout, you can't use <<bit-ints-be,big-endian
+bit integers>> here without splitting the `b` value into 2 separate
+members.
+
+This is because each byte in a *big-endian* bit field is gradually "filled"
+with members from the most significant bit (7) to the least significant (0),
+and if the current byte is filled up to LSB, the parsing continues on
+MSB of next byte. It follows that `b` really can't be represented
+with a single attribute using this order, because `c` and `a` are standing
+in the way.
+
+Little-endian bit fields use the reversed parsing direction: bytes are filled
```

Comment:
Some offtop:

IMHO this point should be redecided, it won't play nice with switcheable endianness (which means that the struct definition must be exactly the same for be and le). Though enumerating the fields from lower bytes to higher looks more natural, KS interface is from higher bits to lower.